### PR TITLE
Should be final commits for Git version parent info PR #598 (#597)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
CC: @achauphan

This should be the remaining commits to address the issues in my latest review of PR:

* #598 

given in https://github.com/TriBITSPub/TriBITS/pull/598#pullrequestreview-1839529322.

This is what is called a ["Stacked PR"](https://bssw.io/blog_posts/pull-request-size-matters).

### CODE REVIEW

File: tribits/core/package_arch/TribitsGlobalMacros.cmake [hunk 1]
```diff
@@ -1215,53 +1211,41 @@ endmacro()
...
     tribits_generate_single_repo_version_string(
-       "${CMAKE_CURRENT_SOURCE_DIR}/${EXTRAREPO_DIR}"
-       SINGLE_REPO_VERSION)
-    string(APPEND REPO_VERSION_FILE_STR
-      "*** Git Repo: ${EXTRAREPO_DIR}\n"
-      "${SINGLE_REPO_VERSION}\n" )
-
-    #print_var(REPO_VERSION_FILE_STR)
+       "${CMAKE_CURRENT_SOURCE_DIR}/${extraRepoDir}" singleRepoVersionStr
+       INCLUDE_COMMIT_PARENTS ${${PROJECT_NAME}_SHOW_GIT_COMMIT_PARENTS})
```
Comment (bartlettroscoe) [inline tribits/core/package_arch/TribitsGlobalMacros.cmake:1239]: @achauphan, note that technically there is no automated test that runs this call with `${PROJECT_NAME}_SHOW_GIT_COMMIT_PARENTS = ON`.  But the lack of this test seems like a low risk given there is just a single argument that changes and everything else is tested.

Comment (achauphan) [reply tribits/core/package_arch/TribitsGlobalMacros.cmake:1239]: @bartlettroscoe, yeah that's a good point. It is only being testing in a single repo case and not in a multi-repo case. Would modifying the test [tribits_add_advanced_test(CTestDriver_TribitsExMetaProj_clone_default_branch_remote)](https://github.com/TriBITSPub/TriBITS/blob/685c8d5fdb94fe591e4dbd1e0687c41943961461/test/ctest_driver/TribitsExampleMetaProject/CMakeLists.txt#L114C1-L114C85) and enabling `${PROJECT_NAME}_SHOW_GIT_COMMIT_PARENTS = ON` be a reasonable method of covering this?

**Review by achauphan — APPROVED** [review]
Thanks for the help with finishing this up Ross. I also really appreciated all your guided comments about best practices when it comes to Git and Cmake! :)

This looks good to me!

**Comment by bartlettroscoe** [comment]
> Would modifying the test [tribits_add_advanced_test(CTestDriver_TribitsExMetaProj_clone_default_branch_remote)](https://github.com/TriBITSPub/TriBITS/blob/685c8d5fdb94fe591e4dbd1e0687c41943961461/test/ctest_driver/TribitsExampleMetaProject/CMakeLists.txt#L114C1-L114C85) and enabling `${PROJECT_NAME}_SHOW_GIT_COMMIT_PARENTS = ON` be a reasonable method of covering this?

@achauphan, that is not a bad idea.  Let me try that locally and see what that looks like. (It looks likes those top commits are merge commits so we should see two parent comments.)

**Comment by bartlettroscoe** [comment]
FYI: I pushed this branch to the main repo so that I could run the GHA testing on it and I manually fired off the GHA jobs and they completed and all passed as shown at:

* https://github.com/TriBITSPub/TriBITS/actions/runs/7647786546

So now I feel good about merging this PR.

**Pull request merged**